src/torchmetrics/functional/image/qnr.py

```diff
@@ -0,0 +1,82 @@
+# Copyright The Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional
+
+from torch import Tensor
+from typing_extensions import Literal
+
+from torchmetrics.functional.image.d_lambda import spectral_distortion_index
+from torchmetrics.functional.image.d_s import spatial_distortion_index
+from torchmetrics.utilities.imports import _TORCHVISION_AVAILABLE
+
+if not _TORCHVISION_AVAILABLE:
+    __doctest_skip__ = ["_quality_with_no_reference_compute", "quality_with_no_reference"]
+
+
+def quality_with_no_reference(
+    preds: Tensor,
+    ms: Tensor,
+    pan: Tensor,
+    pan_lr: Optional[Tensor] = None,
+    alpha: float = 1,
+    beta: float = 1,
+    norm_order: int = 1,
+    window_size: int = 7,
+    reduction: Literal["elementwise_mean", "sum", "none"] = "elementwise_mean",
+) -> Tensor:
+    """Calculate `Quality with No Reference`_ (QualityWithNoReference_) also known as QNR.
+
+    Metric is used to compare the joint spectral and spatial distortion between two images.
+
+    Args:
+        preds: High resolution multispectral image.
+        ms: Low resolution multispectral image.
+        pan: High resolution panchromatic image.
+        pan_lr: Low resolution panchromatic image.
```

Comment:
can you include in the docstring how it's different when `pan_lr` is passed vs None?

Reply:
@justusschock this metric inherits properties from `D_lambda` and `D_S`. for the description about `pan_lr`, what do u think about updating the docstring in `D_S` in this PR as well?